File: 1899. Merge Triplets to Form Target Triplet.py
class Solution:
    def mergeTriplets(self, triplets: List[List[int]], target: List[int]) -> bool:
        # T : O(m * n)
        if target in triplets:
            return True
        trip = [False] * len(target)
        for t in triplets:
            if t[0] > target[0] or t[1] > target[1] or t[2] > target[2]:
                continue
            
            for i, v in enumerate(t):
                if v == target[i]:
                    trip[i] = True
            if all(trip):
                return True

        return False

# I am a noob 
class Solution:
    def mergeTriplets(self, triplets: List[List[int]], target: List[int]) -> bool:
        # T : O(len(triplets * len(target)))
        # Remove the triplets with values greater than target 
        if target in triplets:
            return True

        toRemove = set()
        for idx,t in enumerate(triplets):
            for i, n in enumerate(t):
                if n > target[i]:
                    toRemove.add(idx)
                    break
        toRemove = sorted(list(toRemove))
        
        # Remove from the last index as size of the triplets array will change.
        for i in range(len(toRemove)-1, -1, -1):
            triplets.pop(toRemove[i])
        
        flags = [False] * len(target)
        for idx, t in enumerate(triplets):
            for i, n in enumerate(t):
                if n == target[i]:
                    flags[i] = True
                if all(flags):
                    return True
        return False

File: 1899.MergeTripletstoFormTargetTriplet.java
class Solution {
    public boolean mergeTriplets(int[][] triplets, int[] target) {
        boolean[] trip = new boolean[target.length];
        for (int i = 0; i < trip.length; i++){
            trip[i] = false;
        }

        for (int[] t : triplets){
            if (t[0] > target[0] || t[1] > target[1] || t[2] > target[2]){
                continue;
            }

            for ( int j = 0; j < t.length; j++){
                if (t[j] == target[j]){
                    trip[j] = true;
                }
            }
            boolean checker = true;
            for ( boolean b : trip){
                if (b == false){
                    checker = false;
                }
            }
            if (checker){
                return true;
            }
        }
        return false;
    }
}
